lib/Basics/process-utils.cpp

```diff
@@ -1029,24 +1030,52 @@ ExternalProcessStatus TRI_CheckExternalProcess(ExternalId pid, bool wait, uint32
 
   if (external->_status == TRI_EXT_RUNNING || external->_status == TRI_EXT_STOPPED) {
 #ifndef _WIN32
-    int opts;
-    int loc = 0;
+    if (timeout > 0) {
+      // if we use a timeout, it means we cannot use blocking
+      wait = false;
+    }
 
+    int opts;
     if (wait) {
       opts = WUNTRACED;
     } else {
       opts = WNOHANG | WUNTRACED;
     }
 
-    TRI_pid_t res = waitpid(external->_pid, &loc, opts);
+    int loc = 0;
+    TRI_pid_t res = 0;
+    bool timeoutHappened = false;
+    if (timeout) {
+      TRI_ASSERT((opts & WNOHANG) != 0);
+      double endTime = 0.0; 
```

Comment:
Why not calculate endTime right there and get rid of check  at line 1057 `if (endTime == 0.0) ?`

Reply:
@jsteemann intend here was to save one systemcall, if the waitpid gives the right result on first call.
I'd rather like to keep it that way, since this is similar to devel.